src/cargo/ops/cargo_compile.rs

```diff
@@ -118,41 +160,8 @@ pub fn compile_pkg<'a>(root_package: &Package,
         return Err(human("jobs must be at least 1"))
     }
 
-    let override_ids = try!(source_ids_from_config(options.config, root_package.root()));
-
-    let (packages, resolve_with_overrides, sources) = {
-        let mut registry = PackageRegistry::new(options.config);
-
-        if let Some(source) = source {
-            registry.add_preloaded(root_package.package_id().source_id(), source);
-        }
-
-        // First, resolve the root_package's *listed* dependencies, as well as
-        // downloading and updating all remotes and such.
-        let resolve = try!(ops::resolve_pkg(&mut registry, root_package));
-
-        // Second, resolve with precisely what we're doing. Filter out
-        // transitive dependencies if necessary, specify features, handle
-        // overrides, etc.
-        let _p = profile::start("resolving w/ overrides...");
-
-        try!(registry.add_overrides(override_ids));
-
-        let method = Method::Required{
-            dev_deps: true, // TODO: remove this option?
-            features: &features,
-            uses_default_features: !no_default_features,
-        };
-
-        let resolved_with_overrides =
-                try!(ops::resolve_with_previous(&mut registry, root_package,
-                                                method, Some(&resolve), None));
-
-        let packages = try!(ops::get_resolved_packages(&resolved_with_overrides,
-                                                       &mut registry));
-
-        (packages, resolved_with_overrides, registry.move_sources())
-    };
+    let (packages, resolve_with_overrides, sources) =
+            try!(resolve_dependencies(root_package, config, source, features, no_default_features));
```

Comment:
Could you be sure to wrap this line to 80 chars? It may be easier to read as:

```rust
let res = try!(resolve_dependencies(...));
let (a, b, c) = res;
```